src/services/StreamService.ts

```diff
@@ -8,9 +8,9 @@ class StreamService {
      * to a JS object
      * @param event
      */
-    public static getActivitiesStream(event: any) {
-        return event.Records.filter((record: DynamoDBRecord) => { // Retrieve "MODIFY" events
-            return record.eventName === "MODIFY";
+    public static getVisitsStream(event: any) {
+        return event.Records.filter((record: DynamoDBRecord) => { // Retrieve "MODIFY" visit events
+            return record.eventName === "MODIFY" && record.dynamodb && record.dynamodb.NewImage && record.dynamodb.NewImage.activityType.S === "visit";
```

Comment:
I really hoped to find a good "optional chaining" option for TS, but apparently not. https://medium.com/openmindonline/js-monday-10-optional-chaining-proposal-d9c42b5da517

Reply:
Yeah the remaining proposed changed could be picked during the typescript migration